src/core/api/emit_seek_events.ts

```diff
@@ -49,18 +48,24 @@ export default function emitSeekEvents(
       return EMPTY;
     }
 
-    const isSeeking$ = observableFromEvent(mediaElement, "seeking")
-      .pipe(mapTo("seeking" as const));
-
-    const hasSeeked$ = observableFromEvent(mediaElement, "seeked").pipe(
+    const isSeeking$ = clock$.pipe(
+      distinctUntilChanged((prev, curr) => prev.stalled === curr.stalled ||
+                                           prev.stalled?.reason === curr.stalled?.reason),
```

Comment:
Why?
I don't think that's a good idea to filtering them that way here.
For example, if there's multiple seek where none triggered a stall (example seeking multiple times in buffered data), we wouldn't get multiple "seeking" events, yet I think we should.

The "seeking" event was made to be as close as possible to the real media element's seeking event. Consequently, I think we should just filter out internal-seeks from it without being extra-zealous.

Reply:
Also I think you're considering all stalling events as seeking here, which is wrong. Buffering is not seeking, nor is loading.

Note that you can just listen to the event that triggered the clock here

Reply:
Also I think there's definitely a problem with the current logic even with all those points fixed:
If we're seeking due to an internal seek yet we are not stalled (because the internal seek was into buffered data), we would still send a "seeking" event.

There's something to change on the clock side IMO. Maybe `internalSeek` has to be moved out of the `stalled` object